src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py

```diff
@@ -1581,10 +1581,22 @@ def do_blackbox_fuzzing(self, fuzzer, fuzzer_directory, job_type):
 
     # Run the fuzzer to generate testcases. If error occurred while trying
     # to run the fuzzer, bail out.
+    testcase_generation_start_time = time.time()
```

Comment:
We don't want to include failures? What if there's exceptions? I think having a context manager can help here and avoid cluttering up the code with metrics code.

Reply:
I don't see how to include failures here. If an exception is thrown on line 1587, then we do not really care about partial successes (and we do not even have a way to get a partial result from the inner function, since it is throwing anyway).

Am I missing something obvious here?

Reply:
Re the context manager, an auxiliary function is simpler IMO, since there is really nothing special to do when exceptions show up

Reply:
Auxiliary sgtm!